Sources/AsyncHTTPClient/Singleton.swift

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2023 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension HTTPClient {
+    /// A globally shared, singleton ``HTTPClient``.
+    ///
+    /// The returned client uses the following settings:
+    /// - configuration is ``HTTPClient/Configuration/browserLike`` (matching the platform's default/prevalent browser as well as possible)
+    /// - `EventLoopGroup` is ``HTTPClient/defaultEventLoopGroup`` (matching the platform default)
+    /// - logging is disabled
+    public static var shared: HTTPClient {
+        return globallySharedHTTPClient
+    }
+}
+
+private let globallySharedHTTPClient: HTTPClient = {
```

Comment:
Is there any reason we can't just have a `static let shared` instead?

Reply:
Fair enough! Not opposed to this, just curious why you didn't do the IMO more obvious `static let`.